packages/adapter/addon/rest.ts

```diff
@@ -998,13 +1034,13 @@ class RESTAdapter extends Adapter.extend(BuildURLMixin) {
       return this._fetchRequest(hash)
         .then(response => {
           _response = response;
-          return determineBodyPromise(response, requestData);
+          return determineBodyPromise(response, requestData as JQueryAjaxSettings);
         })
-        .then(payload => {
+        .then((payload: Payload) => {
           if (_response.ok && !(payload instanceof Error)) {
-            return fetchSuccessHandler(adapter, payload, _response, requestData);
+            return fetchSuccessHandler(adapter, payload, _response, requestData as FetchRequestData);
```

Comment:
Why do we need casts here

Reply:
You make a good point.  I [reworked](https://github.com/emberjs/data/pull/7358/commits/44b02c63c3f3a24fd23df566405476c714a731b6#diff-932693a0d32639d2ef4116b0d4060d31d063f287c74b53bd03466b4a2eb9407fR1038) this.  Cast'ed it because requestData was either for fetch or jquery.  Now each declaration of `requestData` is in it's own conditional and typed accordingly.